src/js/reducers/conversation-reducer.js

```diff
@@ -20,7 +20,12 @@ const sortMessages = (messages) => messages.sort((a, b) => {
 });
 
 const addMessage = (messages, message) => {
-    const existingMessage = messages.find((m) => m._id === message._id);
+    let existingMessage = messages.find((m) => m._id === message._id);
+
+    // let's try to match against recently sent messages instead
+    if (!existingMessage) {
```

Comment:
Could check for `role === 'appUser'` here as a small optimization. App maker messages that come in through faye shouldn't suffer from this bug